### DISCUSSION TITLE
Sign v2: Lower session expiry time

### DISCUSSION THREAD

**Discussion by roxaneletourneau** [WalletConnect/walletconnect-monorepo#1600] [Developer Support]
> WalletConnect v2.0 introduces expiry timestamps for sessions which are currently defaulted to 7 days.

Using the `@walletconnect/sign-client` is it possible to lower the expiry time? If so, how can it be achieved?

**Comment by bkrem** [comment]
Hi @roxaneletourneau,

Currently SignClient v2 only provides the `signClient.extend({ topic: sessionTopic })` method to extend an existing session by the default duration (7 days), but not to set custom expiry values for a session.

Could you (and anyone else interested in this functionality) explain your use case for lower session expiries? :pray: That way the WalletConnect team can make an informed decision across platforms regarding possible support for this.

**Reply by roxaneletourneau** [reply]
Hi @bkrem, thank you for your answer. I was looking for this feature for testing purposes. Being able to set a low timeout so we can easily test dapp/wallet behavior on session expiry.

**Reply by llbartekll** [reply]
hey @roxaneletourneau  what about using session delete..? should fit your needs!
